### ISSUE TITLE
uninitialized constant Aws::S3::Errors

### ISSUE THREAD

**Issue by tschaub** [travis-ci/dpl#413]
We're seeing failures on a number of travis-ci.com builds with the following trace:

```
/usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.14/lib/dpl/provider/s3.rb:74:in `rescue in deploy': uninitialized constant Aws::S3::Errors (NameError)
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.14/lib/dpl/provider/s3.rb:73:in `deploy'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.14/lib/dpl/cli.rb:32:in `run'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.14/lib/dpl/cli.rb:7:in `run'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.14/bin/dpl:5:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
```

This looks somewhat similar to #325

**Comment by tschaub** [comment]
Here's the difference between a good and bad build:

```diff
--- good.txt    2016-03-01 16:48:30.000000000 -0700
+++ bad.txt     2016-03-01 16:48:44.000000000 -0700
@@ -1,12 +1,12 @@
 Installing deploy dependencies
 Fetching: jmespath-1.1.3.gem (100%)
 Successfully installed jmespath-1.1.3
-Fetching: aws-sdk-core-2.2.20.gem (100%)
-Successfully installed aws-sdk-core-2.2.20
-Fetching: aws-sdk-resources-2.2.20.gem (100%)
-Successfully installed aws-sdk-resources-2.2.20
-Fetching: aws-sdk-2.2.20.gem (100%)
-Successfully installed aws-sdk-2.2.20
+Fetching: aws-sdk-core-2.2.21.gem (100%)
+Successfully installed aws-sdk-core-2.2.21
+Fetching: aws-sdk-resources-2.2.21.gem (100%)
+Successfully installed aws-sdk-resources-2.2.21
+Fetching: aws-sdk-2.2.21.gem (100%)
+Successfully installed aws-sdk-2.2.21
 4 gems installed
 Fetching: mime-types-2.99.1.gem (100%)
 Successfully installed mime-types-2.99.1
```

**Comment by henrikhodne** [comment]
Looks like a bad version of the `aws-sdk` gem. 2.2.22 is released now, and 2.2.21 is yanked, so I think a restart of the build should work.

**Comment by tschaub** [comment]
Looks like things are working again with the 2.2.22 gems.  Thanks @henrikhodne.
